Sources/Utility/PkgConfig.swift

```diff
@@ -23,9 +23,13 @@ public enum PkgConfigError: Swift.Error {
 /// This is needed because on Linux machines, the search paths can be different
 /// from the standard locations that we are currently searching.
 private let pkgConfigSearchPaths: [AbsolutePath] = {
-    let searchPaths = try? Process.checkNonZeroExit(
-        args: "pkg-config", "--variable", "pc_path", "pkg-config").chomp()
-    return searchPaths?.split(separator: ":").map({ AbsolutePath(String($0)) }) ?? []
+    if let searchPaths = try? Process.checkNonZeroExit(
+        args: "pkg-config", "--variable", "pc_path", "pkg-config").chomp() {
+        return searchPaths.split(separator: ":").map({ AbsolutePath(String($0)) })
+    } else {
+        stderrStream <<< "Warning: Could not find the program 'pkg-config' on your system"
```

Comment:
Take a look at `DiagnosticEngine`. We usually use that to emit diagnostics so we can also easily unit test them e.g.: https://github.com/apple/swift-package-manager/blob/master/Sources/PackageGraph/PackageGraphLoader.swift#L103

Reply:
I can take the `DiagnosticEngine` all the way up from `SwiftTool` and pass it all the way down to the `PkgConfig` initializer, and then pass it into `pkgConfigSearchPaths` (making it a lazily computed property). Two questions:

1) That seemed like a long way to string the engine through to make it available, is that approach correct?
2) There's a `static` function using `pkgConfigSearchPaths` [here](https://github.com/apple/swift-package-manager/blob/master/Sources/Utility/PkgConfig.swift#L118). Should I remove it as `static` or might there be a better way to bring it down?

Thanks!